### PR TITLE
ENH OpenMP w/ clang for OSX

### LINKED ISSUE #1: ENH handle OpenMP on linux and mac osx
So right now, the OpenMP support between linux and osx is spotty.
1. OSX has no support for OpenMP.
2. If the proper libraries are not in the linker path for OpenMP on linux, then the recipe will fail.

I think the easiest way to fix this would be to compile with conda's `gcc` for osx only and then include `libgcc` as a run dependence. 

See [this](https://github.com/conda-forge/staged-recipes/pull/2628) PR for osx and this [comment](https://github.com/conda-forge/staged-recipes/pull/2573#discussion_r107445878) from @jakirkham.

### PR DESCRIPTION
**Pull request by beckermr**
Closes #1 

Is an alternative to #2

### CODE REVIEW

**Comment by jakirkham** [comment]
Will need to add `libcxx` as a dependency too.

**Comment by beckermr** [comment]
Thanks! I got it to compile, but the linking is off. :(

**Comment by beckermr** [comment]
Mmmmk. @jakirkham Take a look and let me know if you think this is good to go!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,11 +24,15 @@ requirements:
     - python
     - setuptools
     - toolchain
+    - libcxx  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Would add `clangdev` for macOS above as that is going to be needed to build with `libcxx` and `openmp`. Though it is getting pulled in automatically ATM. That probably won't be the case going forward.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,11 +24,15 @@ requirements:
...
+    - openmp  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Would pin these like so.

```yaml
     - libcxx 4.0.*  # [osx]
     - openmp 4.0.*  # [osx]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -24,11 +24,15 @@ requirements:
...
 
   run:
     - python
     - numpy
     - scipy
+    - libcxx  # [osx]
+    - openmp  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Also would pin these like so.

```yaml
     - libcxx 4.0.*  # [osx]
     - openmp 4.0.*  # [osx]
```

**Comment by jakirkham** [comment]
Sorry this fell off the radar. Have added a couple of comments above. Some of these were determined relatively recently like the pinnings.

**Comment by beckermr** [comment]
I will make another or and merge.

**Comment by gwerbin** [comment]
Does this work on Linux or just MacOS?

**Comment by beckermr** [comment]
It should work on all systems except windows.

**Pull request merged**